From 4c9c2be310b2b196b9f693834f6f9be62fa2ec0f Mon Sep 17 00:00:00 2001
From: nferre <nferre@50fbe906-d41e-0410-8a96-31537896a350>
Date: Wed, 14 Oct 2009 13:31:51 +0000
Subject: [PATCH] at91: extend power management to at91sam9g45 chip

Add at91sam9g45 defines to select proper RAM chip for power management
handling.

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>

git-svn-id: svn://rfolxts01.rfo.atmel.com/at91_sandbox/linux-2.6.x/branches/linux-2.6.30-at91@11619 50fbe906-d41e-0410-8a96-31537896a350
---
 arch/arm/mach-at91/pm.c           |    6 ++++--
 arch/arm/mach-at91/pm_slowclock.S |   11 +++++++----
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-at91/pm.c b/arch/arm/mach-at91/pm.c
index 242ba6e..2ec022d 100644
--- a/arch/arm/mach-at91/pm.c
+++ b/arch/arm/mach-at91/pm.c
@@ -40,9 +40,11 @@
 #define sdram_selfrefresh_enable()	at91_sys_write(AT91_SDRAMC_SRR, 1)
 #define sdram_selfrefresh_disable()	do {} while (0)
 
-#elif defined(CONFIG_ARCH_AT91CAP9)
+#elif defined(CONFIG_ARCH_AT91CAP9) || defined(CONFIG_ARCH_AT91SAM9G45)
 #include <mach/at91cap9_ddrsdr.h>
-
+#if defined(CONFIG_ARCH_AT91SAM9G45)
+#define AT91_DDRSDRC	AT91_DDRSDRC1
+#endif
 static u32 saved_lpr;
 
 static inline void sdram_selfrefresh_enable(void)
diff --git a/arch/arm/mach-at91/pm_slowclock.S b/arch/arm/mach-at91/pm_slowclock.S
index 987fab3..b29807a 100644
--- a/arch/arm/mach-at91/pm_slowclock.S
+++ b/arch/arm/mach-at91/pm_slowclock.S
@@ -18,7 +18,7 @@
 
 #ifdef CONFIG_ARCH_AT91RM9200
 #include <mach/at91rm9200_mc.h>
-#elif defined(CONFIG_ARCH_AT91CAP9)
+#elif defined(CONFIG_ARCH_AT91CAP9) || defined(CONFIG_ARCH_AT91SAM9G45)
 #include <mach/at91cap9_ddrsdr.h>
 #else
 #include <mach/at91sam9_sdramc.h>
@@ -34,6 +34,9 @@
 #warning Assuming EB1 SDRAM controller is *NOT* used
 #endif
 
+#ifdef CONFIG_ARCH_AT91SAM9G45
+#define AT91_DDRSDRC	AT91_DDRSDRC1
+#endif
 /*
  * When SLOWDOWN_MASTER_CLOCK is defined we will also slow down the Master
  * clock during suspend by adjusting its prescalar and divisor.
@@ -127,7 +130,7 @@ ENTRY(at91_slow_clock)
 	/* Put SDRAM in self-refresh mode */
 	mov	r3, #1
 	str	r3, [r2, #AT91_SDRAMC_SRR]
-#elif defined(CONFIG_ARCH_AT91CAP9)
+#elif defined(CONFIG_ARCH_AT91CAP9) || defined(CONFIG_ARCH_AT91SAM9G45)
 	/* Enable SDRAM self-refresh mode */
 	ldr	r3, [r2, #AT91_DDRSDRC_LPR - AT91_DDRSDRC]
 	str	r3, .saved_sam9_lpr
@@ -239,7 +242,7 @@ ENTRY(at91_slow_clock)
 
 #ifdef CONFIG_ARCH_AT91RM9200
 	/* Do nothing - self-refresh is automatically disabled. */
-#elif defined(CONFIG_ARCH_AT91CAP9)
+#elif defined(CONFIG_ARCH_AT91CAP9) || defined(CONFIG_ARCH_AT91SAM9G45)
 	/* Restore LPR on AT91CAP9 */
 	ldr	r3, .saved_sam9_lpr
 	str	r3, [r2, #AT91_DDRSDRC_LPR - AT91_DDRSDRC]
@@ -271,7 +274,7 @@ ENTRY(at91_slow_clock)
 #ifdef CONFIG_ARCH_AT91RM9200
 .at91_va_base_sdramc:
 	.word AT91_VA_BASE_SYS
-#elif defined(CONFIG_ARCH_AT91CAP9)
+#elif defined(CONFIG_ARCH_AT91CAP9) || defined(CONFIG_ARCH_AT91SAM9G45)
 .at91_va_base_sdramc:
 	.word AT91_VA_BASE_SYS + AT91_DDRSDRC
 #else
-- 
1.5.6.5

